operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -248,41 +251,42 @@ func (r *ReconcileIstioOperator) Reconcile(request reconcile.Request) (reconcile
 		err := r.client.Update(context.TODO(), iop)
 		if err != nil {
 			if errors.IsNotFound(err) {
-				log.Infof("Could not add finalizer to %v: the object was deleted", request)
+				scope.Infof("Could not add finalizer to %s: the object was deleted.", iopName)
 				return reconcile.Result{}, nil
 			} else if errors.IsConflict(err) {
-				log.Infof("Could not add finalizer to %v due to conflict. Operation will be retried in next reconcile attempt", request)
+				scope.Infof("Could not add finalizer to %s due to conflict. Operation will be retried in next reconcile attempt.", iopName)
 				return reconcile.Result{}, nil
 			}
-			log.Errorf("Failed to update IstioOperator with finalizer, %v", err)
+			scope.Errorf(errdict.OperatorFailedToAddFinalizer, "Failed to add finalizer to IstioOperator CR %s: %s", iopName, err)
```

Comment:
So this will log:

`Failed to add finalizer to IstioOperator CR foobar: some error message moreInfo=This error occurred because a finalizer could not be added to the IstioOperator CR. The controller uses the finalizer to temporarily prevent the CR from being deleted while the Istio control plane is being deleted. impact=The error indicates that when the IstioOperator CR is deleted, the Istio control plane may not be fully removed. action=Delete and re-add the IstioOperator CR. If this error persists, contact support if using managed Istio. Otherwise check if the issue is already fixed in another version of Istio at https://github.com/istio/istio/issues, raise an issue, or ask for help in discuss.istio.io. likelyCause=a problem with the k8s API server`

Did we need 112 words to express this? It seems a bit repetative

Reply:
yeah i agree, per your other comments i'm trimming things back. but keep in mind that this is an error which should be an exceptional condition. if it's spamming repeatedly, that should be fixed. otherwise i haven't ever heard users complaining that an error message was too verbose :-)